### PR TITLE
feat: Add ImageGrid ui component.

### PR DESCRIPTION
**Pull request by liamcho**
"The cat is crying because the earth is dying 😿"

Screen Shot 2023-08-10 at 6 07 48 PM

### CODE REVIEW

File: src/ui/ImageGrid/index.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
+@import '../../styles/variables';
+
+.sendbird-text {
+  color: var(--primary-1);
+}
```
Comment (sravan-s) [inline src/ui/ImageGrid/index.scss:5]: you should delete this CSS rule

File: src/ui/ImageGrid/__tests__/ImageGrid.spec.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import ImageGrid from "../index";
+
+describe('ImageGrid', () => {
```
Comment (sravan-s) [inline src/ui/ImageGrid/__tests__/ImageGrid.spec.js:6]: If you run `npm t` in local,

File: src/utils/numberToPx.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default (value: string | number): string => {
+  return typeof value === 'number' ? `${value}px` : value;
```
Comment (sravan-s) [inline src/utils/numberToPx.ts:2]: 👍

File: src/ui/ImageGrid/__tests__/ImageGrid.spec.js (hunk 2 above)
Comment (sravan-s) [reply src/ui/ImageGrid/__tests__/ImageGrid.spec.js:6]: *will explain layers of testing

File: src/ui/ImageGrid/index.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      display: 'flex',
+      justifyContent: 'flex-end',
+    }}>
+      <div
+        className='sendbird-image-grid'
+        style={{
+          gap,
+          padding: gap,
+          borderRadius,
+        }}
```
Comment (sravan-s) [inline src/ui/ImageGrid/index.tsx:35]: discuss this markup

File: src/ui/ImageGrid/index.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+const DEFAULT_BORDER_RADIUS = 12;
+
+export default function ImageGrid({
+  items,
+  gap = DEFAULT_GAP,
+  borderRadius = DEFAULT_BORDER_RADIUS,
+}: ImageGridProps): ReactElement {
+
+  gap = numberToPx(gap);
+  borderRadius = numberToPx(borderRadius);
```
Comment (sravan-s) [inline src/ui/ImageGrid/index.tsx:22]: discuss mutating props

File: src/ui/ImageGrid/index.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    }}>
+      <div
+        className='sendbird-image-grid'
+        style={{
+          gap,
+          padding: gap,
+          borderRadius,
+        }}
+      >
+        { items }
```
Comment (sravan-s) [inline src/ui/ImageGrid/index.tsx:37]: maybe use children

File: src/ui/ImageRenderer/index.tsx [hunk 7]
```diff
@@ -112,7 +114,7 @@ const ImageRenderer = ({
                 backgroundPosition: 'center',
                 backgroundSize: 'cover',
                 backgroundImage: `url(${url})`,
-                borderRadius: circle ? '50%' : null,
+                borderRadius: circle ? '50%' : borderRadius,
```
Comment (sravan-s) [inline src/ui/ImageRenderer/index.tsx:117]: maybe move logic outside ~

File: src/ui/ImageGrid/index.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+}: ImageGridProps): ReactElement {
+
+  gap = numberToPx(gap);
+  borderRadius = numberToPx(borderRadius);
+
+  return (
+    <div style={{
+      display: 'flex',
+      justifyContent: 'flex-end',
+    }}>
```
Comment (sravan-s) [inline src/ui/ImageGrid/index.tsx:28]: add classname -> move those styles to the class
`sendbird-image--grid-wrap`

File: src/ui/ImageGrid/index.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  size?: string | number;
+  gap?: string | number;
+  borderRadius?: string | number;
+}
+
+const DEFAULT_GAP = 4;
+const DEFAULT_BORDER_RADIUS = 12;
+
+export default function ImageGrid({
+  items,
```
Comment (sravan-s) [inline src/ui/ImageGrid/index.tsx:16]: use -> `children` or `renderItems`

Comment (sravan-s) [reply src/ui/ImageGrid/index.tsx:16]: children
```
<ImageGrid>
  <>
    <img1 />
    <img2 />
 </>
</ImageGrid>
```

Comment (sravan-s) [reply src/ui/ImageGrid/index.tsx:16]: `<ImageGrid renderItems={} />`

**Review by sravan-s — APPROVED** [review]

**Review by sravan-s — CHANGES_REQUESTED** [review]
Apply changes

File: src/ui/ImageGrid/index.tsx (hunk 9 above)
Comment (liamcho) [reply src/ui/ImageGrid/index.tsx:16]: Below works
```
<ImageGrid>
  { items }
</ImageGrid>
```

**Comment by liamcho** [comment]
@sravan-s applied all previous feeds.

**Review by sravan-s — APPROVED** [review]

**Review by sravan-s — CHANGES_REQUESTED** [review]
Fix lint before merging

**Review by sravan-s — APPROVED** [review]

**Pull request merged**